apps/docs/src/i18n/index.md

```diff
@@ -0,0 +1,115 @@
+---
+outline: [2, 3]
+---
+
+<script lang="ts" setup>
+import { data } from './languages.data';
+</script>
+
+# Internationalization
+
+Onyx supports internationalization out-of-the-box. This includes commonly used translations
+that are used by our Onyx components like texts for cancel / confirm buttons or "No data" scenarios.
+
+::: tip Default translations
+English is always supported so you don't need to do anything if you only need English texts.
+:::
+
+## Build-in languages
+
+<ul>
+  <li v-for="language in data.languages" :key="language">
+    {{ language.name }} ({{ Math.round(language.keyCount / data.totalKeys * 100) }}% translated)
+  </li>
+</ul>
+
+::: details Which texts are translated?
+Below you can find all translatable texts that are used by our components.
+
+<<< ../../../../packages/sit-onyx/src/i18n/locales/en-US.json
+:::
+
+## Usage
+
+You need to globally provide the i18n instance for Onyx. We assume that you are already using some third-party library such as `vue-i18n`
+for managing your app translations so Onyx will integrate nicely into your setup.
+
+::: code-group
+
+```vue [App.vue]
+<script lang="ts" setup>
+import { provideI18n } from "sit-onyx";
+import deDE from "sit-onyx/locales/de-DE.json";
+import { useI18n } from "vue-i18n";
+
+const { locale } = useI18n();
+
+provideI18n({
+  // The Onyx locale will be updated whenever your vue-i18n locale changes
+  locale,
+  // make sure that the key for each language is the same that you are using
+  // for your vue-i18n JSON files
+  messages: { "de-DE": deDE },
```

Comment:
Currently not because you directly import the `.json` files so there is no index file for them. But the project should be able to achieve this with native Vite import like so:
```ts
const languages = import.meta.glob("sit-onyx/locales/*.json", { import: "default", eager: true });
```

Reply:
can't we do just that for them inside a file in the /locales folder and export the full locales, in terms of DX? ^^

Reply:
Is there a use case where a project will import ALL languages? I'd prefer to not use this because then you don't have tree-shaking

Reply:
DIA e.g. already reached 28 language files, I know it will take some time until we have that much in onyx included but it's not unrealistic if the projects support :)

Reply:
thx!

Reply:
alternatively, using a function with a parameter to pass over only the relevant languages might help?

Reply:
A function would still not have tree shaking because Vite can not statically analyse it.
If so many locales are imported, we should support lazy loading them anyways.
I'd like to delay this feature until we have so many build-in locales that it makes sense to implement it :)

Reply:
ok, should we create a follow-up ticket or just keep it in mind? ^^

Reply:
I created a draft ticket, but we probably need a few months before we get to that point^^
https://github.com/orgs/SchwarzIT/projects/5/views/7?pane=issue&itemId=49970134